File: app/src/main/java/com/boardgamegeek/sorter/PlayersSorter.java
package com.boardgamegeek.sorter;

import android.content.Context;
import android.database.Cursor;
import android.support.annotation.NonNull;

import com.boardgamegeek.provider.BggContract;
import com.boardgamegeek.provider.BggContract.PlayPlayers;

public abstract class PlayersSorter extends Sorter {

	public PlayersSorter(@NonNull Context context) {
		super(context);
	}

	@Override
	protected String getDefaultSort() {
		return PlayPlayers.NAME + BggContract.COLLATE_NOCASE;
	}

	public String getDisplayInfo(Cursor cursor) {
		return getHeaderText(cursor);
	}
}

File: app/src/main/java/com/boardgamegeek/sorter/PlayersSorter.kt
package com.boardgamegeek.sorter

import android.content.Context
import com.boardgamegeek.provider.BggContract
import com.boardgamegeek.provider.BggContract.PlayPlayers
import com.boardgamegeek.ui.model.Player

abstract class PlayersSorter(context: Context) : Sorter(context) {

    override val defaultSort: String
        get() = PlayPlayers.NAME + BggContract.COLLATE_NOCASE

    abstract fun getSectionText(player: Player): String

    open fun getDisplayText(player: Player): String? = null
}
